File: plugins/bindump.py
import tomllib
import logging


class Plugin:
    config: dict[str, str] = {}

    def __init__(self, config: str) -> None:
        self.config = tomllib.loads(config)

    def exec(self, data: bytes) -> None:
        logging.info(f"Dumping {len(data)} bytes to {self.config['path']}")
        with open(self.config["path"], "ba") as f:
            f.write(data)

File: plugins/bindump.lua
return {
    name = "bindump",
    on_rx = function(data)
        local file = io.open("bindump", "ab")
        local log = BUILD_LOGGER("bindump")
        if (file ~= nil) then
            local size = 0
            io.output(file)
            for i, d in ipairs(data) do
                io.write(string.char(d))
                size = size + 1;
            end
            log.info(string.format("Written %d bytes", size))
            io.flush()
        else
            log.error("Cannot open output file")
        end
    end
}
